pkg/certificates/certs.go

```diff
@@ -70,36 +71,20 @@ func createCACertTemplate(expirationInterval time.Duration) (*x509.Certificate,
 	return rootCert, nil
 }
 
-// Create cert template that we can use on the server for TLS
-func createServerCertTemplate(expirationInterval time.Duration) (*x509.Certificate, error) {
-	serverCert, err := createCertTemplate(expirationInterval)
-	if err != nil {
-		return nil, err
-	}
-	serverCert.KeyUsage = x509.KeyUsageDigitalSignature
-	serverCert.ExtKeyUsage = []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth}
-	serverCert.Subject = pkix.Name{
-		Organization: []string{Organization},
-		// Do not use the same value with CA.
-		CommonName: "control-plane-server",
-	}
-	return serverCert, err
-}
-
-// Create cert template that we can use on the client for TLS
-func createClientCertTemplate(expirationInterval time.Duration) (*x509.Certificate, error) {
-	clientCert, err := createCertTemplate(expirationInterval)
+// Create cert template that we can use on the client/server for TLS
+func createClientServerCertTemplate(expirationInterval time.Duration, san string) (*x509.Certificate, error) {
+	cert, err := createCertTemplate(expirationInterval, san)
 	if err != nil {
 		return nil, err
 	}
-	clientCert.KeyUsage = x509.KeyUsageDigitalSignature
-	clientCert.ExtKeyUsage = []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth}
-	clientCert.Subject = pkix.Name{
+	cert.KeyUsage = x509.KeyUsageDigitalSignature
+	cert.ExtKeyUsage = []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth, x509.ExtKeyUsageClientAuth}
+	cert.Subject = pkix.Name{
 		Organization: []string{Organization},
 		// Do not use the same value with CA.
-		CommonName: "control-plane-client",
+		CommonName: "control-plane-client-server",
```

Comment:
Should we put in the SAN for the CommonName?

If not (and we're not worried about changing the CN), I might remove `control-plane` from the CN, but this is also fine.

Reply:
This is really a nit, and isn't super-important either way.